### PR TITLE
[DUOS-2447][risk=no] Create Study on V3 Dataset Create Endpoint

### PR DESCRIPTION
**Pull request by connorlbark**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2447

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by kaitlynoflaherty — APPROVED** [review]
👍

File: src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java [hunk 1]
```diff
@@ -177,6 +196,21 @@ private List<FileStorageObject> uploadFiles(Map<String, FormDataBodyPart> files,
 
     }
 
+    private List<FileStorageObject> uploadFilesForStudy(Map<String, FormDataBodyPart> files,
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java:218]: ~Minor naming nit. This method is extracting a list of Study FSOs from the mass of supplied form data. Something like `extractStudyFiles` might be more descriptive.~

Edit: I misread this at first, it's doing two things, extraction and upload.

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java:218]: I think these private methods could use some javadoc.

File: src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java [hunk 2]
```diff
@@ -246,197 +279,222 @@ Optional<DatasetProperty> extract(DatasetRegistrationSchemaV1 registration, Cons
...
-            new DatasetPropertyExtractor(
-                    "PI Name", "piName", PropertyType.String,
-                    (registration, consentGroup) -> registration.getPiName()),
-            new DatasetPropertyExtractor(
-                    "Study Name", "studyName", PropertyType.String,
-                    (registration, consentGroup) -> registration.getStudyName()),
-            new DatasetPropertyExtractor(
-                    "Study Type", "studyType", PropertyType.String,
-                    (registration, consentGroup) -> {
+    public record StudyPropertyExtractor(
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java:304]: Can we pull this out into its own class?

**Review by rushtong — APPROVED** [review]
Looks really good 👍🏽 Comments inline are optional.

File: src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java (hunk 2 above)
Comment (connorlbark) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetRegistrationService.java:304]: I am honestly not super happy with this approach. I want to rethink it - it's overly complex just to save some lines of code. I feel like we should keep it as-is and reconsider later refactoring it. I'll put in a comment to do that as a follow up, but I feel like it's low-priority (personally).

**Review by fboulnois — APPROVED** [review]
lgtm

**Pull request merged**